.github/workflows/main.yml

```diff
@@ -0,0 +1,66 @@
+name: Main workflow
+
+on:
+  pull_request:
+  push:
+  schedule:
+    # Prime the caches every Monday
+    - cron: 0 1 * * MON
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - macos-latest
+          - ubuntu-latest
+        ocaml-compiler:
+          - 5.0.x
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Use OCaml ${{ matrix.ocaml-compiler }}
+        uses: ocaml/setup-ocaml@v2
+        with:
+          ocaml-compiler: ${{ matrix.ocaml-compiler }}
+
+      - run: opam install . --deps-only --with-test
+
+      - run: opam exec -- dune build
+
+      - run: opam exec -- dune runtest
+
+  lint-fmt:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Use OCaml 5.0.x
+        uses: ocaml/setup-ocaml@v2
+        with:
+          ocaml-compiler: 5.0.x
```

Comment:
Why do you have a hard-coded 5.0.x compiler here instead of using the matrix component like above?

This is my last comment. Once that's answered/changed, let's merge and try it!

Reply:
But I can't get it to work this way, can you please check the syntax @pitag-ha !
It was working previously when I had simple ubuntu-os and running the command

Reply:
Could you try running the formatting/linting jobs only on linux? So concretely, I mean going back to hard-coding `linux-latest` for the `runs-on` field of the formatting/linting jobs, but using the matrix for the `ocaml-compiler` fields.

Reply:
so should I revert back two commits?

Reply:
I copied code to online yaml syntax checkers, but they don't provide the solutions.
I'll try to find a vscode extension, or google again

Reply:
Yes, it was ubuntu latest and ocaml 5.0.x

Reply:
I think it should run now